doc/source/data/execution-configurations.rst

```diff
@@ -0,0 +1,54 @@
+.. _execution_configurations:
+
+=================================
+Execution Configurations
+=================================
+
+Ray Data provides a number of configurations that can be used to control various aspects
+of Ray Dataset execution. These configurations can be modified by the user using
+:class:`~ray.data.ExecutionOptions` and :class:`~ray.data.DataContext`. 
+This guide describes the most important of these configurations and when to use them.
+
+Configuring :class:`~ray.data.ExecutionOptions`
+===============================================
+
+The :class:`~ray.data.ExecutionOptions` class is used to configure options during Ray Dataset execution.
+To use it, you can modify the attributes in the current :class:`~ray.data.DataContext` object's `execution_options`. For example:
+
+.. code-block::
+
+   ctx = ray.data.DataContext.get_current()
+   ctx.execution_options.verbose_progress = True
+
+* `resource_limits`: Set a soft limit on the resource usage during execution. Auto-detected by default.
+* `exclude_resources`: Amount of resources to exclude from Ray Data. Set this if you have other workloads running on the same cluster. Note: 
+
+  * If using Ray Data with Ray Train, training resources are automatically excluded. Otherwise, off by default.
+  * For each resource type, ``resource_limits`` and ``exclude_resources`` can not be both set.
+
+* `locality_with_output`: Set this to prefer running tasks on the same node as the output node (node driving the execution). It can also be set to a list of node ids to spread the outputs across those nodes. Off by default.
+* `preserve_order`: Set this to preserve the ordering between blocks processed by operators under the streaming executor. Off by default.
+* `actor_locality_enabled`: Whether to enable locality-aware task dispatch to actors. This parameter applies to both stateful :meth:`~ray.data.Dataset.map` and :meth:`~ray.data.Dataset.streaming_split` operations. Off by default.
+* `verbose_progress`: Whether to report progress individually per operator. By default, only AllToAll operators and global progress is reported. This option is useful for performance debugging. On by default.
+
+For more details on each of the preceding options, see the API documentation for :class:`~ray.data.ExecutionOptions`.
+
+Configuring :class:`~ray.data.DataContext`
+==========================================
+The :class:`~ray.data.DataContext` class is used to configure more general options for Ray Data usage, such as observability/logging options,
+error handling/retry behavior, and internal data formats. To use it, you can modify the attributes in the current :class:`~ray.data.DataContext` object. For example:
```

Comment:
Wdym by internal data formats?

Reply:
this is referring to things like using pandas block type, setting target min/max block size, using polars, etc. added some clarification